From 270366db1337552649351346961f7c2fd8af3219 Mon Sep 17 00:00:00 2001
From: Ash Charles <ashcharles@gmail.com>
Date: Thu, 5 Jun 2014 13:42:21 -0700
Subject: [PATCH] Cast back to EGLNativeDisplay type.

Signed-off-by: Ash Charles <ashcharles@gmail.com>
---
 src/platform_x11_generic.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/platform_x11_generic.cpp b/src/platform_x11_generic.cpp
index 609bef4..430033b 100644
--- a/src/platform_x11_generic.cpp
+++ b/src/platform_x11_generic.cpp
@@ -101,12 +101,12 @@ native_display::native_display(char const *p_name)
 	if (p_name == NULL)
 	{
 		std::cerr << "Using default X11 display\n";
-		m_egl_native_display = XOpenDisplay(0);
+		m_egl_native_display = reinterpret_cast < EGLNativeDisplayType > (XOpenDisplay(0));
 	}
 	else
 	{
 		std::cerr << "Using X11 display " << p_name << "\n";
-		m_egl_native_display = XOpenDisplay(p_name);
+		m_egl_native_display = reinterpret_cast < EGLNativeDisplayType> (XOpenDisplay(p_name));
 	}
 	if (m_egl_native_display == NULL)
 		std::cerr << "Could not open display\n";
-- 
1.8.3.2

