From e1733405320d7043e3a1e7d15b12bc88a6c598fa Mon Sep 17 00:00:00 2001
From: "Andrew C. Smith" <acsmith@gumstix.com>
Date: Sat, 14 Dec 2019 14:23:29 -0500
Subject: [PATCH] Hard code the gstreamer pipeline to use nvarguscamerasrc.

---
 utils/camera.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/utils/camera.py b/utils/camera.py
index e7424a3..585c6d2 100644
--- a/utils/camera.py
+++ b/utils/camera.py
@@ -77,7 +77,8 @@ def open_cam_usb(dev, width, height):
 
 def open_cam_onboard(width, height):
     """Open the Jetson onboard camera."""
-    gst_elements = str(subprocess.check_output('gst-inspect-1.0'))
+    #gst_elements = str(subprocess.check_output('gst-inspect-1.0'))
+    gst_elements = 'nvarguscamerasrc' 
     if 'nvcamerasrc' in gst_elements:
         # On versions of L4T prior to 28.1, you might need to add
         # 'flip-method=2' into gst_str below.
-- 
2.7.4

